src/test/clojure/clara/test_rules.clj

```diff
@@ -1183,75 +1183,85 @@
                                  [?t <- Temperature (= ?l location)]
                                  [:not [Cold (= temperature (:temperature ?t))]]]]])
 
-        s (mk-session [cold-not-match-temp] :cache false)
-        s-with-prior (mk-session [negation-with-prior-bindings] :cache false)
-        s-with-nested (mk-session [nested-negation-with-prior-bindings] :cache false)]
+        object-query (dsl/parse-query [] [[?o <- Object]])
 
-    (is (= [{}]
-           (-> s
-               (fire-rules)
-               (query cold-not-match-temp))))
+        s (mk-session [cold-not-match-temp object-query] :cache false)
+        s-with-prior (mk-session [negation-with-prior-bindings object-query] :cache false)
+        s-with-nested (mk-session [nested-negation-with-prior-bindings object-query] :cache false)
+
+        no-system-types? (fn [session]
```

Comment:
This is testing the wrapping of the ancestors-fn, namely to make sure that no internal implementation details leak out to the user in the form of NegationResult matching on their productions.  The other way to do this would be to explicitly assert on an enumerated list of all Object instances in the session after the test - maybe we should do that but it doesn't seem ideal either.